### PR TITLE
Add support for Circle (https://www.circle-lang.org/)

### PR DESCRIPTION
**Pull request by thedmd**
PR adds support for `circle` compiler, implementing C++ with `@meta` extensions [circle-lang](https://www.circle-lang.org/).

It is based on C++ language support. I tested locally all build options ("Execute the code", "Compile to binary").
Support for Intel ASM syntax will arrive in next build of the circle compiler.

All `@meta` keywords are listed, but not highlighted properly by Monaco yet.

### CODE REVIEW

File: static/modes/cppcircle-mode.js
Comment (mattgodbolt) [inline static/modes/cppcircle-mode.js:62]: I'll go play with the editor to see why the `@` keywords aren't working. ( https://microsoft.github.io/monaco-editor/monarch.html if you want to try too! )

**Review by mattgodbolt — CHANGES_REQUESTED** [review]
Looking great, some small things. Did we lost the circle.amazon.properties along the way? I'm sure there was something there. Don't worry either way as we can create one easily enough :) (and it's hard to test. FWIW I symlink `circle.local.properties` to `circle.amazon.properties` to test when I play around locally)

File: lib/compilers/circle.js
Comment (partouf) [inline lib/compilers/circle.js:1]: It's not 2018

File: etc/config/circle.defaults.properties
Comment (thedmd) [reply etc/config/circle.defaults.properties:11]: Sure

File: etc/config/circle.defaults.properties
Comment (thedmd) [reply etc/config/circle.defaults.properties:7]: This I don't know. I installed it to `/opt/circle-lang`. I found no defaults anywhere. I can point defaults there and `amazon` and `local` to `compiler-explorer/circle-lang`.

File: etc/config/circle.defaults.properties
Comment (thedmd) [reply etc/config/circle.defaults.properties]: Done

File: etc/config/circle.defaults.properties
Comment (thedmd) [reply etc/config/circle.defaults.properties]: Done

File: etc/config/circle.defaults.properties
Comment (thedmd) [reply etc/config/circle.defaults.properties]: Done

File: examples/circle/default.cpp
Comment (thedmd) [reply examples/circle/default.cpp]: Done

File: etc/config/c++.amazon.properties
Comment (partouf) [inline etc/config/c++.amazon.properties:325]: Wait, wasn't this correct? "based on Herb Sutter [his] paper", right? https://grammar.yourdictionary.com/punctuation/apostrophe-rules.html 
The apostrophe could also be behind the name, but it should be there afaik?

File: static/modes/cppcircle-mode.js
Comment (thedmd) [reply static/modes/cppcircle-mode.js:28]: Done

File: static/modes/cppcircle-mode.js
Comment (thedmd) [reply static/modes/cppcircle-mode.js]: Done

File: lib/compilers/circle.js
Comment (thedmd) [reply lib/compilers/circle.js]: Indeed. Fixed.

File: etc/config/c++.amazon.properties
Comment (thedmd) [reply etc/config/c++.amazon.properties]: My mistake. I shouldn't touch this file at all.

(I removed it 'temporary' because it boke highlighing in editor.) Reverted.

File: static/modes/cppcircle-mode.js
Comment (thedmd) [reply static/modes/cppcircle-mode.js]: Done

File: static/modes/cppcircle-mode.js
Comment (thedmd) [reply static/modes/cppcircle-mode.js:45]: Hack does work. Thanks!
```js
    // Hack to put `@...` keywords in place
    cppcircle.tokenizer.root.unshift(
        [/@\w+/, {cases: {'@keywords': {token: 'keyword.$0'}}}]
    );
```

File: etc/config/circle.amazon.properties
Comment (mattgodbolt) [inline etc/config/circle.amazon.properties:2]: Thanks! I think I'll tweak these a bit rather than rely on the groups in the defaults; but I won't block merging on this.

File: lib/compilers/circle.js
Comment (mattgodbolt) [inline lib/compilers/circle.js:35]: Should this be commented back in?

File: static/modes/cppcircle-mode.js
Comment (mattgodbolt) [reply static/modes/cppcircle-mode.js:28]: Thanks: we can drop the `cpp` entirely so we pick up any other changes in the `cppp`. I'll happily make that tweak after merging.

**Review by mattgodbolt — APPROVED** [review]
Looks great! Thanks so mcuh :) On equestion but I'll merge and we can tweak afterwards as needed!

**Comment by mattgodbolt** [comment]
This is now live!

**Pull request merged**
